src/tiff.ml

```diff
@@ -920,25 +928,37 @@ module Data = struct
 
       (* We iterate through every strip *)
       for strip = first_strip to last_strip - 1 do
-        let raw_strip_buffer = Cstruct.create strip_bytecounts.(strip) in
-        let strip_offset = Optint.Int63.of_int strip_offsets.(strip) in
-
-        (* Fill the strip buffer *)
-        ro ~file_offset:strip_offset [ raw_strip_buffer ];
-
-        let expected_size = rows_per_strip * width * bytes_per_pixel in
-
         let strip_buffer =
-          match Ifd.compression ifd with
-          | No_compression ->
-              if Cstruct.length raw_strip_buffer < expected_size then
-                failwith "Strip is unexpectedly short";
-              raw_strip_buffer
-          | LZW ->
-              let uncompressed_buffer = Cstruct.create expected_size in
-              Lzw.decode raw_strip_buffer uncompressed_buffer;
-              uncompressed_buffer
-          | _ -> failwith "Unsupported compression"
+          match List.assoc_opt strip t.strip_cache with
```

Comment:
I wonder if we're better-off using something with O(1) lookup here. It probably doesn't matter too much but we do incur a bit of a hit the first time we load something in and have the caching policy set (as we have to loop through the entire strip_cache for every strip).